solutionbox/code_free_ml/mltoolbox/code_free_ml/data/local_predict.py

```diff
@@ -0,0 +1,251 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Runs local prediction on a trained model."""
+
+
+import argparse
+import os
+import six
+import sys
+import json
+
+import tensorflow as tf
+from tensorflow.python.lib.io import file_io
+
+
+def parse_arguments(argv):
+  """Parse command line arguments.
+  Args:
+    argv: includes the script's name.
+  Returns:
+    argparse object
+  """
+  parser = argparse.ArgumentParser(
+      description='Runs local prediction')
+  # I/O args
+  parser.add_argument('--predict-data',
+                      required=True,
+                      help='Data to run prediction on')
+  parser.add_argument('--trained-model-dir',
+                      required=True,
+                      help='Usually train_output_path/model.')
+  parser.add_argument('--output-location',
+                      required=True,
+                      help=('Location to save output.'))
+  parser.add_argument('--output-format',
```

Comment:
Do we also need to support output-file-name? Otherwise for example in Datalab, I have to look for "predictions-xxxxx-of-xxxxx" files, and the "predictions" keyword need to be documented and cannot be changed. In the case of no sharding, I would expect to use output-file-name without appending 'xxxxx-of-xxxxx'.